lib/Backend/GlobOpt.cpp

```diff
@@ -6935,20 +6959,34 @@ GlobOpt::SetSymStore(ValueInfo *valueInfo, Sym *sym)
     }
     if (valueInfo->GetSymStore() == nullptr || valueInfo->GetSymStore()->IsPropertySym())
     {
-        valueInfo->SetSymStore(sym);
+        SetSymStoreDirect(valueInfo, sym);
     }
 
     return sym;
 }
 
 void
+GlobOpt::SetSymStoreDirect(ValueInfo * valueInfo, Sym * sym)
```

Comment:
Direct here is no type check in the original SetSymStore, but all sym store need check changed bit. Cannot overload SetSymStore since both have identical input parameter types.

---
In reply to: [68327056](https://github.com/Microsoft/ChakraCore/pull/1152#discussion_r68327056) [](ancestors = 68327056)